examples/external_request/with_harvester.rb

```diff
@@ -0,0 +1,82 @@
+require 'net/http'
+require 'bundler'
+Bundler.require
+
+require "new_relic/telemetry_sdk"
+
+unless ENV["API_KEY"]
+  raise "No License key supplied.  Export API_KEY environment variable!" 
+end
+
+def setup_buffer_harvesting common_attributes = {test_attribute: 'example'}
+  @harvester = NewRelic::TelemetrySdk::Harvester.new 
+  @span_client = NewRelic::TelemetrySdk::SpanClient.new
+  @buffer = NewRelic::TelemetrySdk::Buffer.new common_attributes
+  @harvester.register 'external_spans', @buffer, @span_client
+  @harvester.start
+end
+
+def stop_harvester
+  @harvester.stop
+end
+
+def random_id length=16
+  length.times.map{rand(16).to_s(16)}.join
+end
+
+def record_external_request
+  start_time = Time.now
+  response = yield
+  finish_time = Time.now
+  duration = finish_time - start_time
+
+  custom_attributes = {
```

Comment:
Consider making the attributes that are the same for each span common attributes.